go/kbfs/libkbfs/folder_block_ops.go

```diff
@@ -954,23 +954,45 @@ func (fbo *folderBlockOps) wrapWithBlockLock(fn func()) dirCacheUndoFn {
 	}
 }
 
+func (fbo *folderBlockOps) obfuscatorFromPathLocked(
+	lState *kbfssync.LockState, dir data.Path) data.Obfuscator {
+	fbo.blockLock.AssertAnyLocked(lState)
+	node := fbo.nodeCache.Get(dir.TailPointer().Ref())
+	if node == nil {
+		// When there is no node corresponding to this path yet, we
+		// make a new obfuscator.  This means that entry names that
+		// are retrieved from the `DirData` derived from this
+		// `Obfuscator` can't be directly compared for equality with
+		// entry names retrieved from other `DirData`s, or input by
+		// outside applications, since the obfuscators won't match.
+		// This is ok as long as we always do critical comparisons and
+		// map lookups with the plaintext string (as in
+		// `DirData.Lookup()`).
```

Comment:
FYI the `DIrData` changes will be coming in the next PR, but I thought I'd split this out for now for clarity of review.